components/Repositories/LotteryFactorChart.tsx

```diff
@@ -0,0 +1,210 @@
+import { HiOutlineTicket } from "react-icons/hi";
+import { useMemo, useState } from "react";
+import Card from "components/atoms/Card/card";
+import Pill from "components/atoms/Pill/pill";
+import { DayRange } from "components/shared/DayRangePicker";
+import humanizeNumber from "lib/utils/humanizeNumber";
+import StackedOwners from "components/Workspaces/StackedOwners";
+import DevProfile from "components/molecules/DevProfile/dev-profile";
+
+type LotteryFactorChartProps = {
+  lotteryFactor: RepositoryLottoFactor | undefined;
+  isLoading: boolean;
+  error: Error | undefined;
+  range: DayRange;
+};
+
+export default function LotteryFactorChart({ lotteryFactor, isLoading, error, range }: LotteryFactorChartProps) {
+  const [hovered, setHovered] = useState<string | undefined>(undefined);
+  const topFourPercentage = useMemo(
+    () =>
+      isLoading || !lotteryFactor
+        ? 0
+        : Number(
+            (
+              lotteryFactor.all_contribs
+                .slice(0, 4)
+                .reduce((prev, contributor) => prev + contributor.percent_of_total, 0) * 100
+            ).toPrecision(1)
+          ),
+    [lotteryFactor]
+  );
```

Comment:
Since only `lotteryFactor` is in the dependency array, that means error or isLoading will always result in 0 because in both those cases it will be `!lotteryFactor`. Is the `isLoading ||` necessary then?

Reply:
added `isLoading` to the dependency array